src/Associativity.cpp

```diff
@@ -96,37 +98,46 @@ class ConvertSelfRef : public IRMutator {
 };
 
 template<typename T>
-bool visit_associative_binary_op(const string &op_x, const string &op_y, Expr x_part,
-                                 Expr lhs, Expr rhs, AssociativeOp &op) {
+std::pair<bool, bool> visit_associative_binary_op(
+        const string &op_x, const string &op_y, Expr x_part,
+        Expr lhs, Expr rhs, AssociativeOp &op) {
     const Variable *var_a = lhs.as<Variable>();
     if (!var_a || (var_a->name != op_x)) {
         debug(4) << "Can't prove associativity of " << T::make(lhs, rhs) << "\n";
-        return false;
+        return std::make_pair(false, false);
```

Comment:
These std::make_pair calls might be cleaner as initializer lists. E.g. 

  return {false, false}